OpenAPITools/openapi-generator
# [Rust-Axum] Fix uuid in header params causing compilation errors

Routes with header parameters with a `format` of `uuid` in the openAPI specification used to cause a compilation error in the resulting Rust Axum code.
This commit fixes the issue by including the correct conversion trait implementation on the condition that at least one header parameter of `format` `uuid` is included in the specification.
Problem details: #18554  

# How to validate:
The faulty behavior can be reproduced like this:
On the current master branch generate a Rust axum library with the specification described in  #18554 .
Then (when you have installed `rustc` and `cargo`, run `cargo check` in the rust workspace containing the generated library. You will get compile errors in the places where the `uuid` is extracted from the request headers.
To verify the fix do the same with the code changes contained in this pull request.
Now cargo check will not report any compilation errors. Note the change in the `src/header.rs` file. A few lines have been added to it.



### PR checklist
 
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
 
@frol @farcaller @richardwhiuk @paladinzh @jacob-pro
